### PR TITLE
Fix/extract lumberjack

### PR DESCRIPTION
**Pull request by ph**
I have extracted the `lumberjack` gem outside of logstash-forwarder at https://github.com/ph/lumberjack/, this would allow us to hopefully iterate more quickly and clearly separate concerns.

The LSF is still using the lumberjack gem to run his integration testing.

When I did the extract of the code I have keep the change history and rewrote commits to point to the appropriate tickets.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,5 +5,6 @@ gem "insist"
 gem "stud"
 gem "fpm"
 gem "pleaserun"
+gem "jls-lumberjack", :github => "elastic/ruby-lumberjack"
```
Comment (jordansissel) [inline Gemfile]: I don't think we need this change, do we? The gem name isn't changing (yet), and if it does, we'll still publish it normally.

**Comment by jordansissel** [comment]
Other than my one comment, LGTM

File: Gemfile (hunk 1 above)
Comment (ph) [inline Gemfile]: I decided to point to the github repo so we can setup jenkins to trigger a LSF build after a lumberjack change? Using `gem "jls-lumberjack"` will only test against the stable release.

Comment (jordansissel) [inline Gemfile]: hmm, let's make a note to make jenkins more flexible in terms of how we can test such things.

I agree with your reasoning. LGTM

**Pull request closed without merging**